### PR TITLE
Initial Localization Script 

### PR DESCRIPTION
**Pull request by ericajlin**
Add initial localization script with one folder only to test end to end localization pipeline before adding all translate-able files.
Starting with Storage Overview because the range of translatable text.

Changes made:
- Add localization toolset to yml files
- Add restoreLoc.cmd, localizationBuild.cmd files as part of localization toolset requirements
- Add extractStrings script to parse template files into resjson with applicable translate-able strings, generate a LocPayload.json file with paths to resjson files + output configurations. Note this script will change once end to end translation for sample folder is complete to include all workbook files.

### CODE REVIEW

File: scripts/extractStrings.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+/** Copied from InsightsPortal to find parameters in strings. TODO: need to handle column name references */
+function findParameterNames(text) {
+  const ValidParameterNameRegex = "[_a-zA-Z\xA0-\uFFFF][_a-zA-Z0-9\xA0-\uFFFF]*";
+  const ValidSpecifierRegex = "[_a-zA-Z0-9\xA0-\uFFFF\\-\\$\\@\\.\\[\\]\\*\\?\\(\\)\\<\\>\\=\\,\\:]*";
+  var _parameterRegex = new RegExp("\{" + ValidParameterNameRegex + "(:" + ValidSpecifierRegex + ")?\}", "g");
+  var params = null;
+  try {
+    params = text.match(_parameterRegex);
+  } catch (error) {
+    console.error("Cannot extract parameter. ", "ERROR: ", e);
```
Comment (vutran01) [inline scripts/extractStrings.js:132]: should this be variable error?

File: scripts/extractStrings.js [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+    // Add LocProject entry
+    locItems.push({
+      "SourceFile": RESJSONOutputPath.concat("\\", resjsonFileName),
+      "CopyOption": "LangIDOnPath",
+      "OutputPath": LCLOutputPath
+    });
+
+    // Write extracted strings to file
+    writeToFileRESJSON(extracted, fileName, RESJSONOutputPath);
```
Comment (vutran01) [inline scripts/extractStrings.js:261]: is there a resource file for each template? the build loc scripts sees to only process on one file

File: scripts/extractStrings.js (hunk 1 above)
Comment (ericajlin) [reply scripts/extractStrings.js:132]: good catch

File: scripts/extractStrings.js [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+}
+
+/** Validates file type. Expected to be either workbook/cohort file or settings/categoryjson file */
+function isValidFileType(filename) {
+  console.log("Found file: ", filename);
+  if (filename.localeCompare(CategoryResourcesFile) == 0 || filename.localeCompare(SettingsFile) == 0) {
+    return true;
+  }
+  var a = filename.split(".");
+  if (a.length === 1 || (a[0] === "" && a.length === 2)) {
```
[the remark this answers is not in the document]
Comment (ericajlin) [reply scripts/extractStrings.js:73]: will fix, thanks

File: scripts/extractStrings.js [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+}
+
+/**
+ * 
+ * SCRIPT MAIN
+ * 
+ */
+
+if (!process.argv[2]) { // path to extract strings from 
+  console.log('ERROR: Workbook path not provided. Please provide the path to the workbook folder.');
```
Comment (yves-chan) [inline scripts/extractStrings.js:203]: nit: console.error

Comment (ericajlin) [reply scripts/extractStrings.js:203]: will change it in the next iteration

**Pull request merged**